### PR TITLE
feat: CodeBuilder push and unshift can format strings

### PR DESCRIPTION
**Pull request by thibaultcha**
- add an example to CodeBuilder
- convert objc/nsurlsession to not use util.format anymore

We could convert more languages to it before merging this.

### CODE REVIEW

File: src/helpers/code-builder.js [hunk 1]
```diff
@@ -35,29 +52,32 @@ CodeBuilder.prototype.buildLine = function (indentationLevel, line) {
...
- * Add a line at the top of current lines with given indentation level
+ * Invoke buildLine() and add the line at the top of current lines
  * @param {number} [indentationLevel=0] Desired level of indentation for this line
  * @param {string} line Line of code
  * @return {this}
  */
-CodeBuilder.prototype.unshift = function (indentationLevel, str) {
-  this.code.unshift(this.buildLine(indentationLevel, str))
+CodeBuilder.prototype.unshift = function () {
+  this.code.unshift(this.buildLine.apply(this, arguments))
```
Comment (ahmadnassri) [inline src/helpers/code-builder.js:68]: nice, much cleaner!

**Comment by thibaultcha** [comment]
Well, I'm not rebasing this lol

**Comment by ahmadnassri** [comment]
#lazy

**Comment by thibaultcha** [comment]
My PR had no conflict when I proposed it!

**Comment by thibaultcha** [comment]
Cool!

**Pull request merged**